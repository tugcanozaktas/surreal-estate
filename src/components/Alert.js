/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import "../styles/Alert.css";
import PropTypes from "prop-types";

function Alert({ message, success }) {
  if (!message) return null;

  return (
    <div className={`Alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
}

export default Alert;

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};
