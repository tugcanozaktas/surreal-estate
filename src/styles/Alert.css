.Alert{
    text-align: center;
    width: 200px;
    height: 30px;
    border-radius: 5px;
}
.alert-success{
    background-color: rgba(105, 194, 105, 0.678);
    border: 1px solid green;
}
.alert-error{
    background-color: rgba(221, 25, 25, 0.784);
    border: 1px solid rgb(128, 0, 0);
}