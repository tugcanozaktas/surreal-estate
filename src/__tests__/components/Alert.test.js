import React from "react";
import { render } from "@testing-library/react";
import Alert from "../../components/Alert";

test("displays error message", () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);

  expect(getByText(/Error/).textContent).toBe("Error!");
  expect(asFragment()).toMatchSnapshot();
});

test("displays success message", () => {
  const { getByText, asFragment } = render(
    <Alert message="Success!!!" success />,
  );

  expect(getByText(/Success/).textContent).toBe("Success!!!");
  expect(asFragment()).toMatchSnapshot();
});

test("does not render if message is empty", () => {
  const { asFragment } = render(<Alert message="" />);
  const alert = asFragment();
  expect(alert).toMatchSnapshot();
});
