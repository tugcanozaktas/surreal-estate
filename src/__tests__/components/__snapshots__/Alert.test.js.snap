// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`displays success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-success"
  >
    Success!!!
  </div>
</DocumentFragment>
`;

exports[`does not render if message is empty 1`] = `<DocumentFragment />`;
